projects/plugins/jetpack/extensions/extended-blocks/paid-blocks/with-upgrade-banner.jsx

```diff
@@ -70,50 +71,28 @@ export default createHigherOrderComponent(
 			setAttributes( { shouldDisplayFrontendBanner: ! hasParentBanner } );
 		}, [ setAttributes, hasParentBanner ] );
 
-		useEffect( () => {
-			const block = document.querySelector( `.wp-block[data-block="${ clientId }"]` );
-			if ( ! block ) {
-				return;
-			}
-
-			let upgradeBannerContainer = document.querySelector(
-				`.wp-block[data-block="${ clientId }"] > .jetpack-block-upgrade-banner-container`
-			);
-			if ( ! upgradeBannerContainer ) {
-				upgradeBannerContainer = document.createElement( 'div' );
-				upgradeBannerContainer.classList.add( 'jetpack-block-upgrade-banner-container' );
-				block.prepend( upgradeBannerContainer );
-			}
-
-			render(
-				<UpgradePlanBanner
-					className={ `is-${ name.replace( /\//, '-' ) }-paid-block` }
-					title={ null }
-					align={ attributes?.align }
-					visible={ isBannerVisible }
-					description={ usableBlocksProps?.description }
-					requiredPlan={ requiredPlan }
-					context={ bannerContext }
-					onRedirect={ () => trackUpgradeClickEvent( trackEventData ) }
-				/>,
-				upgradeBannerContainer
-			);
-		}, [
-			attributes?.align,
-			clientId,
-			isBannerVisible,
-			name,
-			requiredPlan,
-			trackEventData,
-			usableBlocksProps?.description,
-		] );
+		const blockProps = useBlockProps();
 
 		return (
 			<PaidBlockProvider
 				onBannerVisibilityChange={ setIsVisible }
 				hasParentBanner={ isBannerVisible }
 			>
-				<BlockEdit { ...props } />
+				<div { ...blockProps }>
```

Comment:
This is duplicating all the attributes of the block, including the ID so the DOM ends up with two elements with the same ID which is not valid HTML:

<img width="646" alt="Screenshot 2022-12-13 at 15 42 32" src="https://user-images.githubusercontent.com/1233880/207364260-cb8d1dfd-8e1e-4314-b5e6-9387e2e69259.png">

Reply:
It also applies custom classes that are intended for the block only, potentially causing some style conflicts:

<img width="931" alt="Screenshot 2022-12-13 at 15 49 50" src="https://user-images.githubusercontent.com/1233880/207366070-e7902743-59c3-4f77-b434-99ac2d062019.png">

Reply:
Nice catch, I didn't notice that it duplicates the id.